src/macaron/slsa_analyzer/provenance/loader.py

```diff
@@ -1,48 +1,75 @@
-# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 - 2024, Oracle and/or its affiliates. All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 """This module contains the loaders for SLSA provenances."""
 
 import base64
+import configparser
 import gzip
+import io
 import json
 import zlib
+from urllib.parse import urlparse
 
+import requests
+
+from macaron.config.defaults import defaults
 from macaron.slsa_analyzer.provenance.intoto import InTotoPayload, validate_intoto_payload
 from macaron.slsa_analyzer.provenance.intoto.errors import LoadIntotoAttestationError, ValidateInTotoPayloadError
 from macaron.util import JsonType
 
 
-def load_provenance_file(filepath: str) -> dict[str, JsonType]:
-    """Load a provenance file and obtain the payload.
+def _try_read_url_link_file(file_content: bytes) -> str | None:
+    parser = configparser.ConfigParser()
+    try:
+        parser.read_file(io.TextIOWrapper(io.BytesIO(file_content), encoding="utf-8"))
+        return parser.get("InternetShortcut", "url", fallback=None)
+    except (configparser.Error, UnicodeDecodeError):
+        return None
 
-    Inside a provenance file is a DSSE envelope containing a base64-encoded
-    provenance JSON payload. See: https://github.com/secure-systems-lab/dsse.
-    If the file is gzipped, it will be transparently decompressed.
 
-    Parameters
-    ----------
-    filepath : str
-        Path to the provenance file.
+def _download_url_file_content(url: str, url_link_hostname_allowlist: list[str]) -> bytes:
+    hostname = urlparse(url).hostname
+    if hostname is None or hostname == "":
+        raise LoadIntotoAttestationError("Cannot resolve URL link file: invalid URL")
+    if hostname not in url_link_hostname_allowlist:
+        raise LoadIntotoAttestationError(
+            "Cannot resolve URL link file: target hostname '" + hostname + "' is not in allowed hostnames."
+        )
 
-    Returns
-    -------
-    dict[str, JsonType]
-        The provenance JSON payload.
+    try:
+        # TODO download size limit?
+        timeout = defaults.getint("downloads", "timeout", fallback=120)
+        response = requests.get(url=url, timeout=timeout)
```

Comment:
We should use the [send_get_http_raw](https://github.com/oracle/macaron/blob/ba3fcb0c84d6727de343c247a3181908fcd78410/src/macaron/util.py#L59) helper function so that we can improve the implementation in one place as explained in this [issue](https://github.com/oracle/macaron/issues/398).

Reply:
Done.